src/module-server/middleware/js.ts

```diff
@@ -102,7 +102,14 @@ export const jsMiddleware = async ({
           import.meta.pleasantestArgs = [...window._pleasantestArgs]
         }`;
         const fileSrc = await fs.readFile(file, 'utf8');
-        const inlineStartIdx = fileSrc.indexOf(code);
+        let EOL = '\n';
+        // Find the first line end (\n) and check if the character before is \r
+        // This tells us whether the file uses \r\n or just \n for line endings.
+        // Using node:os.EOL is not sufficient because git on windows
+        // Can be configured to check out files with either kind of line ending.
+        const firstLineEndPos = fileSrc.indexOf('\n');
+        if (fileSrc[firstLineEndPos - 1] === '\r') EOL = '\r\n';
+        const inlineStartIdx = fileSrc.indexOf(code.replaceAll('\n', EOL));
```

Comment:
I wonder if some of this should be pulled out into a helper called `normalizeLineEndings` or `determineLineEndings` or something?